zowe/zowe-explorer-vscode
# tests: Behavior-driven integration and e2e tests using WebdriverIO

## Proposed changes

This pull request implements the following items:
- [x] Behavior-driven integration test bed using WebdriverIO framework and CucumberJS
- [x] Test-driven integration test bed using vscode-test and Mocha
  - Old tests were removed from this test bed, but its still in place to offer flexibility and enable stubbing where needed
- [x] End-to-end (e2e) test bed using WebdriverIO and CucumberJS
  - Screenshots for end-to-end failing test steps (thanks @t1m0thyj 🎉)
- [x] Documentation for how to set up and run tests (both e2e and integration)
- [x] Launch configurations for running e2e and integration tests
- [x] Initial foundation/list of e2e and integration tests as listed in #2322 

## How to test

- Checkout the `tests/wdio-impl` branch in the ZE repo
- Run `pnpm install && pnpm build` at the root of the repo
- Refer to the steps in `packages/zowe-explorer/__tests__/__e2e__/README.md` to prepare your environment for end-to-end testing
  - Once the environment is prepared, run `cd packages/zowe-explorer && pnpm tests:e2e` to run tests
- Refer to the steps in `packages/zowe-explorer/__tests__/__integration__/README.md` to ensure your environment is ready for integration tests
  - Once the environment is prepared, run `cd packages/zowe-explorer && pnpm tests:integration` to run tests

Both test scripts should succeed provided that:
- the user's Zowe config has at least 3 profiles that can be accessed during the integration tests
- the environment variables defined in the `__e2e__/.env` file refer to valid filters and items on the test system during the e2e tests

In addition, notice the new step in the build stage on Ubuntu/Node 20 titled "Integration tests" - these should be passing with the latest changes on this branch.

## Release Notes




Milestone: v3

Changelog:

- Added behavior-driven integration tests and end-to-end tests to verify extension behavior and catch issues during Zowe Explorer development. [#2322](https://github.com/zowe/zowe-explorer-vscode/issues/2322)

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_


## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_


## Further comments

I noticed that we still had a few lint errors in the existing unit tests. I've corrected those as part of this branch.
